mutation_partition.cc

```diff
@@ -1876,7 +1876,7 @@ bool row_marker::compact_and_expire(tombstone tomb, gc_clock::time_point now,
         _timestamp = api::missing_timestamp;
```

Comment:
(commenting here because I can't comment on the description)

A good commit description also explains why the logic was invalid and how it is fixed.

Reply:
I reworded the description, hope that it is more complete and clear now.

Reply:
Much better. Previously I had to guess and go look at the other `is_expired()` implementations to see why this one was invalid.